libgearman/run.cc

```diff
@@ -269,7 +269,8 @@ gearman_return_t _client_run_task(Task *task)
       else if (task->recv->command == GEARMAN_COMMAND_STATUS_RES_UNIQUE)
       {
         task->error_code(GEARMAN_SUCCESS);
-        strncpy(task->unique, task->recv->arg[0], GEARMAN_MAX_UNIQUE_SIZE);
+        memcpy(task->unique, task->recv->arg[0], GEARMAN_MAX_UNIQUE_SIZE);
+        task->unique[GEARMAN_MAX_UNIQUE_SIZE -1]= '\0';
```

Comment:
The new code reads memory past the zero terminator of the source string. As such, it may cause rare protection faults.